--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,5 +1,5 @@
 
-ADD_SUBDIRECTORY( package-manager )
+##ADD_SUBDIRECTORY( package-manager )
 
 INSTALL( FILES notify-message DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/zypp" )
 
--- a/zypp.conf
+++ b/zypp.conf
@@ -217,6 +217,7 @@
 ##			some sane default.
 ##
 ## commit.downloadMode =
+commit.downloadMode = DownloadInAdvance
 
 ##
 ## Defining directory which contains vendor description files.
@@ -326,6 +327,7 @@
 ##	empty
 ##
 # multiversion = provides:multiversion(kernel)
+multiversion = kernel,kernel-adaptation-connext,kernel-adaptation-intel-automotive,kernel-adaptation-medfield,kernel-adaptation-mrst,kernel-adaptation-oaktrail
 
 ##
 ## Path to locks file. If not exist then is create.
--- a/zypp/CMakeLists.txt
+++ b/zypp/CMakeLists.txt
@@ -3,6 +3,7 @@
 ####################################################################
 
 ADD_DEFINITIONS(-DLOCALEDIR="${CMAKE_INSTALL_PREFIX}/share/locale" -DTEXTDOMAIN="zypp" )
+ADD_DEFINITIONS(-DNO_HAL)
 
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 #FILE(WRITE filename "message to write"... )
