Index: libzypp-9.8.3/zypp/CMakeLists.txt
===================================================================
--- libzypp-9.8.3.orig/zypp/CMakeLists.txt
+++ libzypp-9.8.3/zypp/CMakeLists.txt
@@ -917,6 +917,7 @@ SET_LOGGROUP( "satsolver" ${zypp_sat_SRC
 #SET_LOGGROUP( "group" ${zypp_pool_SRCS} )
 
 
+list( REVERSE zypp_lib_SRCS )
 ADD_LIBRARY(zypp SHARED ${zypp_lib_SRCS})
 SET_TARGET_PROPERTIES( zypp PROPERTIES VERSION "${LIBZYPP_VERSION_INFO}" )
 SET_TARGET_PROPERTIES( zypp PROPERTIES SOVERSION "${LIBZYPP_SOVERSION_INFO}" )
