diff -uNr libzypp-8.1.0/zypp/target/TargetImpl.cc libzypp-8.1.0.new/zypp/target/TargetImpl.cc
--- libzypp-8.1.0/zypp/target/TargetImpl.cc	2010-08-06 00:32:16.000000000 +0800
+++ libzypp-8.1.0.new/zypp/target/TargetImpl.cc	2010-11-06 17:52:02.356580841 +0800
@@ -785,7 +785,9 @@
         if ( ! _root.empty() )
           cmd << " -r '" << _root << "'";
 
-        cmd << " -p '" << Pathname::assertprefix( _root, "/etc/products.d" ) << "'";
+        PathInfo productsdir( Pathname::assertprefix( _root, "/etc/products.d" ) );
+        if ( productsdir.isExist() )
+          cmd << " -p '" << productsdir.path() << "'";
 
         if ( ! oldSolvFile.empty() )
           cmd << " '" << oldSolvFile << "'";
