diff -uNr libzypp-9.8.3/zypp/repo/RepoProvideFile.cc libzypp-9.8.3.new/zypp/repo/RepoProvideFile.cc
--- libzypp-9.8.3/zypp/repo/RepoProvideFile.cc	2011-07-04 22:04:11.000000000 +0800
+++ libzypp-9.8.3.new/zypp/repo/RepoProvideFile.cc	2011-07-22 16:37:51.059506789 +0800
@@ -249,6 +249,7 @@
                                               const ProvideFilePolicy & policy_r )
     {
       MIL << loc_r << endl;
+      int count = 0;
       // Arrange DownloadFileReportHack to recieve the source::DownloadFileReport
       // and redirect download progress triggers to call the ProvideFilePolicy
       // callback.
@@ -346,6 +347,11 @@
 
           repo_excpt.remember(e);
 
+          if (count < 3)
+          {
+            count++;
+            --it;
+          }
           WAR << "Trying next url" << endl;
           continue;
         }
