diff -uNr libzypp-8.10.2/zypp/target/TargetImpl.cc libzypp-8.10.2.new/zypp/target/TargetImpl.cc
--- libzypp-8.10.2/zypp/target/TargetImpl.cc	2011-03-17 11:45:57.403456661 +0800
+++ libzypp-8.10.2.new/zypp/target/TargetImpl.cc	2011-03-17 11:45:03.891768999 +0800
@@ -38,12 +38,14 @@
 #include "zypp/ExternalProgram.h"
 #include "zypp/Repository.h"
 
+#include "zypp/FileChecker.h"
 #include "zypp/ResFilters.h"
 #include "zypp/HistoryLog.h"
 #include "zypp/target/TargetImpl.h"
 #include "zypp/target/TargetCallbackReceiver.h"
 #include "zypp/target/rpm/librpmDb.h"
 #include "zypp/target/CommitPackageCache.h"
+#include "zypp/target/rpm/RpmDb.h"
 
 #include "zypp/parser/ProductFileReader.h"
 
@@ -1301,6 +1303,22 @@
 
             try
             {
+              if ( !policy_r.rpmNoSignature() ) 
+              {
+                target::rpm::RpmDb::checkPackageResult res;
+                Target_Ptr target( getZYpp()->getTarget() );
+                res = target->rpmDb().checkPackage(localfile);
+
+                if (res == target::rpm::RpmDb::CHK_OK)
+                  MIL << "Signature is OK" <<endl;
+                else if (res == target::rpm::RpmDb::CHK_NOTFOUND)
+                  ZYPP_THROW( FileCheckException( localfile.value().asString() + " Signature is unknown type" ) );
+                else if (res == target::rpm::RpmDb::CHK_FAIL)
+                  ZYPP_THROW( FileCheckException( localfile.value().asString() + " Signature does not verify" ) );
+                else if (res == target::rpm::RpmDb::CHK_NOKEY)
+                  ZYPP_THROW( FileCheckException( localfile.value().asString() + " Public key is unavailable" ) );
+              }
+
               progress.tryLevel( target::rpm::InstallResolvableReport::RPM_NODEPS_FORCE );
               rpm().installPackage( localfile, flags );
               HistoryLog().install(*it);
@@ -1319,6 +1337,10 @@
                 success = true;
               }
             }
+            catch ( const FileCheckException &e )
+            {
+              ZYPP_THROW(e);
+            }
             catch ( Exception & excpt_r )
             {
               ZYPP_CAUGHT(excpt_r);
