Index: libzypp-8.1.0/zypp/PublicKey.cc
===================================================================
--- libzypp-8.1.0.orig/zypp/PublicKey.cc
+++ libzypp-8.1.0/zypp/PublicKey.cc
@@ -157,7 +157,7 @@ namespace zypp
         static filesystem::TmpDir dir;
         const char* argv[] =
         {
-          "gpg",
+          "gpg2",
           "-v",
           "--no-default-keyring",
           "--fixed-list-mode",
